src/MovingAverage.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Moving Average Filter (linear FIR filter with equal weights)
+ */
+#include "iarc7_sensors/MovingAverage.hpp"
+
+#include <ros/ros.h>
+
+namespace iarc7_sensors {
+
+MovingAverage::MovingAverage(const ros::Duration& time) : filter_time_constant_(time) {}
+
+void MovingAverage::updateFilter(double value, double derivative, const ros::Time& time) {
```

Comment:
That's actually a very simple change to make, it should be able to do this with any data type that has the + operator and multiplication by a scalar.  I think I'd like to do that, because then we can run moving averages on any tensors we want.  I guess that means it has to be a template